core/modules/qdisp/JobDescription.h

```diff
@@ -58,12 +59,13 @@ class ResponseHandler;
 class JobDescription {
 public:
     using Ptr =  std::shared_ptr<JobDescription>;
-    static JobDescription::Ptr create(QueryId qId, int jobId, ResourceUnit const& resource,
+    static JobDescription::Ptr create(qmeta::CzarId czarId, QueryId qId, int jobId,
```

Comment:
Since you're modifying the class then could you also declare the deleted copy constructor and the assignment operator for the class?
```
JobDescription(JobDescription const&) = delete;
JobDescription& operator=(JobDescription const&) = delete;
```